fluent-bit/fluent-bit-go-loki/loki.go

```diff
@@ -0,0 +1,99 @@
+package main
+
+import (
+	"fmt"
+	"os"
+	"regexp"
+	"strconv"
+	"time"
+
+	"github.com/cortexproject/cortex/pkg/util/flagext"
+	"github.com/go-kit/kit/log"
+	"github.com/go-kit/kit/log/level"
+	"github.com/grafana/loki/pkg/logql"
+	"github.com/prometheus/common/model"
+	"github.com/weaveworks/common/logging"
+)
+
+type lokiConfig struct {
+	url        flagext.URLValue
+	batchWait  time.Duration
+	batchSize  int
+	labelSet   model.LabelSet
+	logLevel   logging.Level
+	removeKeys []string
+}
+
+func getLokiConfig(url, batchWait, batchSize, labels, logLevelVal, removeKeyStr string) (*lokiConfig, error) {
+	lc := &lokiConfig{}
+	var clientURL flagext.URLValue
+	if url == "" {
+		url = "http://localhost:3100/api/prom/push"
+	}
+	err := clientURL.Set(url)
+	if err != nil {
+		return nil, fmt.Errorf("Failed to parse client URL")
+	}
+	lc.url = clientURL
+
+	batchWaitValue, err := strconv.Atoi(batchWait)
+	if err != nil || batchWait == "" {
+		batchWaitValue = 1
+	}
+	lc.batchWait = time.Duration(batchWaitValue) * time.Second
+
+	batchSizeValue, err := strconv.Atoi(batchSize)
+	if err != nil || batchSize == "" {
+		batchSizeValue = 100 * 1024
+	}
+	lc.batchSize = batchSizeValue
+
+	if labels == "" {
+		labels = `{job="fluent-bit"}`
+	}
+
+	ls, err := logql.ParseExpr(labels)
```

Comment:
you can use the new https://github.com/grafana/loki/blob/master/pkg/logql/parser.go#L47 `logql.ParseMatchers`